app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerFragment.kt

```diff
@@ -29,58 +28,67 @@ import android.view.LayoutInflater
 import com.infomaniak.mail.MatomoMail.trackSyncAutoConfigEvent
 import com.infomaniak.mail.R
+import com.infomaniak.mail.data.cache.mailboxContent.FolderController
+import com.infomaniak.mail.data.models.Folder
+import com.infomaniak.mail.data.models.Quotas
+import com.infomaniak.mail.data.models.mailbox.Mailbox
+import com.infomaniak.mail.data.models.mailbox.MailboxPermissions
 import com.infomaniak.mail.databinding.FragmentMenuDrawerBinding
 import com.infomaniak.mail.ui.MainActivity
-import com.infomaniak.mail.ui.main.MailboxListFragment
+import com.infomaniak.mail.ui.MainViewModel
+import com.infomaniak.mail.ui.alertDialogs.InputAlertDialog
+import com.infomaniak.mail.ui.bottomSheetDialogs.LockedMailboxBottomSheetDialogArgs
+import com.infomaniak.mail.ui.main.InvalidPasswordFragmentArgs
 import com.infomaniak.mail.ui.main.folder.ThreadListFragmentDirections
 import com.infomaniak.mail.utils.AccountUtils
 import com.infomaniak.mail.utils.ConfettiUtils
-import com.infomaniak.mail.utils.ConfettiUtils.ConfettiType
+import com.infomaniak.mail.utils.ConfettiUtils.ConfettiType.INFOMANIAK
 import com.infomaniak.mail.utils.Utils
 import com.infomaniak.mail.utils.Utils.Shortcuts
-import com.infomaniak.mail.utils.Utils.runCatchingRealm
+import com.infomaniak.mail.utils.extensions.bindAlertToViewLifecycle
 import com.infomaniak.mail.utils.extensions.checkForFolderCreationErrors
 import com.infomaniak.mail.utils.extensions.launchSyncAutoConfigActivityForResult
 import com.infomaniak.mail.utils.extensions.observeNotNull
-import com.infomaniak.mail.utils.extensions.toggleChevron
 import dagger.hilt.android.AndroidEntryPoint
 import kotlinx.coroutines.launch
+import javax.inject.Inject
 
 @AndroidEntryPoint
-class MenuDrawerFragment : MenuFoldersFragment(), MailboxListFragment {
+class MenuDrawerFragment : Fragment() {
 
     private var _binding: FragmentMenuDrawerBinding? = null
     private val binding get() = _binding!! // This property is only valid between onCreateView and onDestroyView
     private val menuDrawerViewModel: MenuDrawerViewModel by viewModels()
+    private val mainViewModel: MainViewModel by activityViewModels()
 
-    var exitDrawer: (() -> Unit)? = null
+    @Inject
+    lateinit var folderController: FolderController
+
+    @Inject
+    lateinit var inputDialog: InputAlertDialog
 
-    override val isInMenuDrawer = true
-    override val defaultFoldersList: RecyclerView by lazy { binding.defaultFoldersList }
-    override val customFoldersList: RecyclerView by lazy { binding.customFoldersList }
+    @Inject
+    lateinit var menuDrawerAdapter: MenuDrawerAdapter
 
-    override val hasValidMailboxes = true
-    override val currentClassName: String = MenuDrawerFragment::class.java.name
-    override val mailboxesAdapter get() = binding.mailboxList.adapter as MailboxesAdapter
+    private val currentClassName: String = MenuDrawerFragment::class.java.name
+
+    var exitDrawer: (() -> Unit)? = null
```

Comment:
Instead of using oddly defined callbacks like this, we should've probably used a SingleLiveEvent inside a shared view model with the activity which is the preferred way of communicating between an Activity and a Fragment I think

Reply:
Ok, I'll assign this to the guy that developed it 2 years ago 😁

Reply:
@LunarX Can you look into that please ? 🤣

Reply:
ℑ'𝔪 𝔱𝔥𝔢 𝔤𝔲𝔶 𝔴𝔥𝔬 𝔡𝔢𝔳𝔢𝔩𝔬𝔭𝔢𝔡 𝔦𝔱 2 𝔶𝔢𝔞𝔯𝔰 𝔞𝔤𝔬 𝔞𝔫𝔡 ℑ 𝔞𝔤𝔯𝔢𝔢 𝔴𝔦𝔱𝔥 𝔱𝔥𝔢 𝔣𝔦𝔯𝔰𝔱 𝔤𝔲𝔶'𝔰 𝔠𝔬𝔪𝔪𝔢𝔫𝔱

Reply:
Ultimately, we don't really need to change that part in this PR, it's mostly just something I noticed since this part moved around

Reply:
I've created a ticket about this.